### PR TITLE
Sudoku(#195)

### PR DESCRIPTION
**Pull request by Sandip75**

### CODE REVIEW

File: Algorithms/Backtracking/Sudoku.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import java.util.Scanner;
```
Comment (vikasgola) [inline Algorithms/Backtracking/Sudoku.java]: Add problem statement

File: Algorithms/Backtracking/Sudoku.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    // Here I will free that number ( choice ). for that i will set choice bit as Zero and all 1. after that i will do & to get back all set bits.
+    
+    int mask = ~(1 << choice);
+		arr[r][c] = 0;
+
+		rows[r] &= mask;
+		cols[c] &= mask;
+		submats[r / 3][c / 3] &= mask;
+	}
+}
```
Comment (vikasgola) [inline Algorithms/Backtracking/Sudoku.java]: Add time and space complexity with one line description.

File: Algorithms/Backtracking/Sudoku.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+		int row = (cellNo - 1) / arr.length;
+		int col = (cellNo - 1) % arr.length;
+    
+    // When value is already present  at any [row][col]
+		if (arr[row][col] != 0) {
+			SolveSudoku(arr, cellNo + 1, rows, cols, submats);
+		} 
+    
+    // When value is zero at any [row][col]
+    else {
```
Comment (vikasgola) [inline Algorithms/Backtracking/Sudoku.java]: Correct your indentation.

File: Algorithms/Backtracking/Sudoku.java (hunk 1 above)
Comment (Sandip75) [reply Algorithms/Backtracking/Sudoku.java]: Solving Sudoku #195

File: Algorithms/Backtracking/Sudoku.java (hunk 3 above)
Comment (Sandip75) [reply Algorithms/Backtracking/Sudoku.java]: ?

File: Algorithms/Backtracking/Sudoku.java (hunk 2 above)
Comment (Sandip75) [reply Algorithms/Backtracking/Sudoku.java]: Yes , I will add it

File: Algorithms/Backtracking/Sudoku.java (hunk 1 above)
Comment (vikasgola) [reply Algorithms/Backtracking/Sudoku.java]: Inside the program in the comments.

**Comment by sahilbansal17** [comment]
@Sandip75 Please make the changes requested by @vikasgola .

**Comment by Sandip75** [comment]
I will change it by tomorrow busy with an intern.

File: Algorithms/Backtracking/Sudoku.java [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+    		int mask = ~(1 << choice);
+		arr[r][c] = 0;
+
+		rows[r] &= mask;
+		cols[c] &= mask;
+		submats[r / 3][c / 3] &= mask;
+	}
+}
+// Time Complexity: O(m^(n^2)), where m is total number of empty cells. At every place there can be n possiblity. 
+// Space Complexity: O(9*3), Since we have used three array of size 9 ( rows[] , cols[] , submats[][] )
```
Comment (vikasgola) [inline Algorithms/Backtracking/Sudoku.java]: This is constant space complexity.
So, you can write it as O(1).

**Pull request merged**